framework/src/mesh/StitchedMesh.C

```diff
@@ -0,0 +1,108 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "StitchedMesh.h"
+#include "Parser.h"
+#include "InputParameters.h"
+
+// libMesh includes
+#include "libmesh/mesh_modification.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<StitchedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName>>("files", "The name of the mesh files to read.  These mesh files will be 'stitched' into the current mesh in this order.");
+
+  params.addRequiredParam<std::vector<BoundaryName>>("stitch_boundaries", "Pairs of boundary names (one after the other) to stitch together for each step.");
```

Comment:
Yes, Cubit should have smarter algorithm on merging.

Another question is that you are stitching mesh one by one, right? How if the third mesh is connected with both the second and the first mesh?